ifrs17-template/Initialization/InitSystemorphUtils.ipynb

```diff
@@ -0,0 +1,170 @@
+            "source": [
+                "# Import Tasks"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeDataNodesAsync(IDataSourceVariable dataSource, string pathToImport, string[] nodes = null){",
+                "\n    nodes = nodes ?? new string[]{\"ES\", \"FR\", \"CH\"};",
+                "\n    var uploads = DataNodeFiles.Where(kvp => nodes.Contains(kvp.Key)).SelectMany(kvp => kvp.Value).ToArray();",
+                "\n    foreach(var pair in uploads) ",
+                "\n        await Import.FromFile(pathToImport + pair.Item1).WithFormat(pair.Item2).WithTarget(dataSource).WithActivityLog().ExecuteAsync();",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeDataNodesAsync(IWorkspace workspace, string pathToImport, string[] nodes = null){",
+                "\n    nodes = nodes ?? new string[]{\"ES\", \"FR\", \"CH\"};",
+                "\n    var uploads = DataNodeFiles.Where(kvp => nodes.Contains(kvp.Key)).SelectMany(kvp => kvp.Value).ToArray();",
+                "\n    foreach(var pair in uploads)",
+                "\n        await Import.FromFile(pathToImport + pair.Item1).WithFormat(pair.Item2).WithTarget(workspace).WithActivityLog().ExecuteAsync();",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeTransactionalDataAsync(IDataSourceVariable dataSource, string pathToImport, string[] nodes = null){",
+                "\n    nodes = nodes ?? new string[]{\"ES\", \"FR\", \"CH\"};",
+                "\n    var uploads = TransactionalFilesByDataNode.Where(kvp => nodes.Contains(kvp.Key)).SelectMany(kvp => kvp.Value).ToArray();",
+                "\n    foreach(var pair in uploads)",
+                "\n        await Import.FromFile(pathToImport + pair.Item1).WithFormat(pair.Item2).WithTarget(dataSource).WithActivityLog().ExecuteAsync();",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeTransactionalDataAsync(IWorkspace workspace, string pathToImport, string[] nodes = null){",
+                "\n    nodes = nodes ?? new string[]{\"ES\", \"FR\", \"CH\"};",
+                "\n    var uploads = TransactionalFilesByDataNode.Where(kvp => nodes.Contains(kvp.Key)).SelectMany(kvp => kvp.Value).ToArray();",
```

Comment:
all these methods have the same code, the dictionary used is the only diff. 
Let's extract the methods that runs the imports

Reply:
done.